common/os/os_util.c

```diff
@@ -391,7 +391,7 @@ str_int_extract(char *str, int *arr, int arr_size, int *num)
 		return (B_FALSE);
 	}
 
-	strncpy(scopy, str, len);
+	strncpy(scopy, str, len + 1);
 	scopy[len] = 0;
```

Comment:
Since `len=strlen(str)` and `scopy` is allocated to be `len+1` bytes, doesn't the line after `strncpy` add the null-terminator always (`scopy[len] = 0`)?

Perhaps replacing all of this with `strdup` would be easier and clearer?